test/merge-coverage.js

```diff
@@ -0,0 +1,245 @@
+ * @returns {boolean}
+ */
+function isCoverageInsufficient(target, actual) {
+  const lineCoverageNotMet = actual.lines.pct < target.lines;
+  const branchCoverageNotMet = actual.branches.pct < target.branches;
+  const functionCoverageNotMet = actual.functions.pct < target.functions;
+  const statementCoverageNotMet = actual.statements.pct < target.statements;
+  return (
+    lineCoverageNotMet ||
+    branchCoverageNotMet ||
+    functionCoverageNotMet ||
+    statementCoverageNotMet
+  );
+}
+
+/**
+ * Checks if the coverage should be bumped up
+ *
+ * @param {CoverageTarget} target
+ * @param {import('istanbul-lib-coverage').CoverageSummaryData} actual
+ * @returns {boolean}
+ */
+function shouldCoverageBeBumped(target, actual) {
+  const lineCoverageNeedsBumped = actual.lines.pct > target.lines + 5;
+  const branchCoverageNeedsBumped = actual.branches.pct > target.branches + 5;
+  const functionCoverageNeedsBumped =
+    actual.functions.pct > target.functions + 5;
+  const statementCoverageNeedsBumped =
+    actual.statements.pct > target.statements + 5;
+  return (
+    lineCoverageNeedsBumped ||
+    branchCoverageNeedsBumped ||
+    functionCoverageNeedsBumped ||
+    statementCoverageNeedsBumped
+  );
+}
+
+/**
+ * Creates and returns a combined coverage summary report of every file in the
+ * provided array.
+ *
+ * @param {string[]} files - array of files generated by fast-glob
+ * @param {libCoverage.CoverageMap} coverageMap
+ * @returns {import('istanbul-lib-coverage').CoverageSummaryData}
+ */
+function getFileCoverage(files, coverageMap) {
+  const subCoverageMap = libCoverage.createCoverageMap({});
+
+  files.forEach((file) => {
+    try {
+      subCoverageMap.merge(
+        coverageMap.fileCoverageFor(`${process.cwd()}/${file}`),
+      );
+    } catch {
+      // If the coverage doesn't exist, it means that it was excluded from
+      // coverage or had no coverage to report, which is fine. Glob is a lot
+      // wider of a net then what the test file runners match against.
+    }
+  });
+
+  const summary = subCoverageMap.getCoverageSummary();
+  return summary;
+}
+
+/**
+ * Checks coverage and reports to console
+ * Throws an error if coverage isn't met
+ *
+ * @param {string} name - The target's name from coverageThresholds in jest
+ *  config
+ * @param {CoverageTarget} target - the target coverage threshold
+ * @param {import('istanbul-lib-coverage').CoverageSummaryData} actual -
+ *  istanbul coverage summary representing actual coverage
+ */
+function checkCoverage(name, target, actual) {
+  const breakdown = generateConsoleReport(target, actual);
+  if (isCoverageInsufficient(target, actual)) {
+    const errorMsg = `Coverage thresholds for ${name} NOT met\n${breakdown}`;
+    throw new Error(errorMsg);
+  } else if (shouldCoverageBeBumped(target, actual)) {
+    const errorMsg = `Coverage thresholds for ${name} EXCEED and need bumped\n${breakdown}`;
```

Comment:
```suggestion
    const errorMsg = `Coverage thresholds for ${name} EXCEEDed and need bumping\n${breakdown}`;
```

Reply:
Agree that the wording is off on my take, but I would prefer something like 

```suggestion
    const errorMsg = `Coverage EXCEEDS threshold for ${name} and must be bumped\n${breakdown}`;
```

Reply:
Agreed yours is better!